### PR TITLE
8315731: Open source several Swing Text related tests

### PR DESCRIPTION
**Pull request by jmtd**
Test-only backport for Oracle parity. 5 newly-opensource Swing tests. Not clean: one required adjusting for JDK<14 (rework instanceof pattern matching). After that, they all pass for 11u.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] [JDK-8315731](https://bugs.openjdk.org/browse/JDK-8315731) needs maintainer approval

### Issue
 * [JDK-8315731](https://bugs.openjdk.org/browse/JDK-8315731): Open source several Swing Text related tests (**Bug** - P4 - Approved)

### Reviewers
 * [Sergey Bylokhov](https://openjdk.org/census#serb) (@mrserb - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk11u-dev.git pull/2434/head:pull/2434` \
`$ git checkout pull/2434`

Update a local copy of the PR: \
`$ git checkout pull/2434` \
`$ git pull https://git.openjdk.org/jdk11u-dev.git pull/2434/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 2434`

View PR using the GUI difftool: \
`$ git pr show -t 2434`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk11u-dev/pull/2434.diff">https://git.openjdk.org/jdk11u-dev/pull/2434.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk11u-dev/pull/2434#issuecomment-1877174141)

### CODE REVIEW

**Comment by jmtd** [comment]
/approval request  Test-only backport for Oracle parity. 5 newly-opensource Swing tests. Not clean: one required adjusting for JDK<14 (rework instanceof pattern matching). After that, they all pass for 11u.

**Comment by GoeLin** [comment]
Please get a review before requesting approval.

**Review by mrserb — APPROVED** [review]

**Comment by jmtd** [comment]
/approval request Test-only backport for Oracle parity. 5 newly-opensource Swing tests. Not clean: one required adjusting for JDK<14 (rework instanceof pattern matching). After that, they all pass for 11u

**Pull request closed without merging**
